homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,320 @@
+    ClimateEntity,
+    ClimateEntityDescription,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, Platform, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+if TYPE_CHECKING:
+    from matter_server.client import MatterClient
+    from matter_server.client.models.node import MatterEndpoint
+
+    from .discovery import MatterEntityInfo
+
+TEMPERATURE_SCALING_FACTOR = 100
+HVAC_SYSTEM_MODE_MAP = {
+    HVACMode.OFF: 0,
+    HVACMode.HEAT_COOL: 1,
+    HVACMode.COOL: 3,
+    HVACMode.HEAT: 4,
+}
+SystemModeEnum = clusters.Thermostat.Enums.ThermostatSystemMode
+ControlSequenceEnum = clusters.Thermostat.Enums.ThermostatControlSequence
+ThermostatFeature = clusters.Thermostat.Bitmaps.ThermostatFeature
+
+
+class ThermostatRunningState(IntEnum):
+    """Thermostat Running State, Matter spec Thermostat 7.33."""
+
+    Heat = 1  # 1 << 0 = 1
+    Cool = 2  # 1 << 1 = 2
+    Fan = 4  # 1 << 2 = 4
+    HeatStage2 = 8  # 1 << 3 = 8
+    CoolStage2 = 16  # 1 << 4 = 16
+    FanStage2 = 32  # 1 << 5 = 32
+    FanStage3 = 64  # 1 << 6 = 64
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter climate platform from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.CLIMATE, async_add_entities)
+
+
+class MatterClimate(MatterEntity, ClimateEntity):
+    """Representation of a Matter climate entity."""
+
+    _attr_temperature_unit: str = UnitOfTemperature.CELSIUS
+    _attr_supported_features: ClimateEntityFeature = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+    )
+    _attr_hvac_mode: HVACMode = HVACMode.OFF
+
+    def __init__(
+        self,
+        matter_client: MatterClient,
+        endpoint: MatterEndpoint,
+        entity_info: MatterEntityInfo,
+    ) -> None:
+        """Initialize the Matter climate entity."""
+        super().__init__(matter_client, endpoint, entity_info)
+
+        # set hvac_modes based on feature map
+        self._attr_hvac_modes: list[HVACMode] = [HVACMode.OFF]
+        feature_map = int(
+            self.get_matter_attribute_value(clusters.Thermostat.Attributes.FeatureMap)
+        )
+        if feature_map & ThermostatFeature.kHeating:
+            self._attr_hvac_modes.append(HVACMode.HEAT)
+        if feature_map & ThermostatFeature.kCooling:
+            self._attr_hvac_modes.append(HVACMode.COOL)
+        if feature_map & ThermostatFeature.kAutoMode:
+            self._attr_hvac_modes.append(HVACMode.HEAT_COOL)
+
+    @property
+    def min_temp(self) -> float:
```

Comment:
I'd move the update of these attributes, max temp, min temp, target temp, from properties to `_update_from_device`.

Reply:
"Min temp" and "max temp" should also be read-only, similar to HVAC modes?

Reply:
They can be dynamic. It's ok to have different max and min temps depending on the mode.

Reply:
I tried to move it, and _update_from_device became very long. I want to ask what the problem is with using property here. If we don't use property and change it to a regular method, can we call it in _update_from_device? Will there be any difference?

Reply:
It's confusing to have to read some logic in the properties and some in the callback. It's better to have all the state update in the same way.

Reply:
all attributes are now updated in the update callback